### PR TITLE
feat(PE-3745): show in progress interactions when editing an ANT

### PR DESCRIPTION
**Pull request by dtfiedler**
Manage modal:
image

image

Manage Table:
image

### CODE REVIEW

File: src/components/modals/ManagePDNTModal/ManagePDNTModal.tsx [hunk 1]
```diff
@@ -69,54 +76,76 @@ function ManagePDNTModal() {
...
       const record = Object.values(pdnsSourceContract.records).find(
-        (r) => r.contractTxId === txId.toString(),
+        (r) => r.contractTxId === contractTxId.toString(),
       );
       const tier = pdnsSourceContract.tiers.history.find(
         (t) => t.id === record?.tier,
       );
       // TODO: add error messages and reload state to row
-      const consolidatedDetails: ManagePDNTRow & any = {
+      const consolidatedDetails: any = {
```
Comment (karlprieb) [inline src/components/modals/ManagePDNTModal/ManagePDNTModal.tsx:103]: why `any`? It would be good to create this `type` or just let TS infer it

Comment (dtfiedler) [reply src/components/modals/ManagePDNTModal/ManagePDNTModal.tsx:103]: we should create a type for it, I'll add 👍

**Comment by dtfiedler** [comment]
while i'm here - i'l add the same thing to undernames table

File: src/state/contexts/GlobalState.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ const initialState: GlobalState = {
     'bLAgYxAdX2Ry-nt6aH2ixgvJXbpsEYm28NgJgyqfs-U',
   ),
   pdnsSourceContract: DEFAULT_PDNS_REGISTRY_STATE,
-  gateway: 'ar-io.dev',
```
Comment (atticusofsparta) [inline src/state/contexts/GlobalState.tsx:21]: Im guessing this is from testing?

Comment (dtfiedler) [reply src/state/contexts/GlobalState.tsx:21]: yeah - although we still havent' seemed to solve the issue with ar-io.dev so we may want to just revert until the changes can be implemented

File: src/utils/transactionUtils/transactionUtils.ts [hunk 3]
```diff
@@ -588,3 +623,36 @@ export function validateTTLSeconds(ttl: number) {
...
+  // find all pending interactions for the contract, find relevant ones related to row attributes
+  const pendingTxRowData = [];
+  for (const i of pendingContractInteractions) {
+    const attributes = getAttributesFromInteractionFunction(i.payload.function);
+    // TODO: this is not pretty, and could be avoided if we rework the ANT contract to allow `setTTL` and `setTransaction` rather than all of them
+    // relying only on setRecord.
+    for (const attribute of attributes) {
+      const nonConfirmedTx = {
+        attribute,
+        value: i.payload[attribute],
```
Comment (atticusofsparta) [inline src/utils/transactionUtils/transactionUtils.ts:645]:
```suggestion
        value: i.payload[attribute === 'targetID' ? 'transactionId' : attribute],
```
We have a mismatching attribute and data key type - IMO the key in the contract should be renamed to 'targetId'. This is a bandaid fix for the moment ^ for the bug where a target ID edit didnt not show up.

File: src/utils/transactionUtils/transactionUtils.ts [hunk 4]
```diff
@@ -588,3 +623,36 @@ export function validateTTLSeconds(ttl: number) {
...
+        id: i.id,
+        valid: i.valid,
+      };
+      if (
+        existingValues[attribute] &&
```
[the remark this answers is not in the document]
Comment (dtfiedler) [reply src/utils/transactionUtils/transactionUtils.ts]: [27369b5](https://github.com/ar-io/pdns-react/pull/214/commits/27369b5c592e166ae1ea6dd9e9808e1d426eb027)

File: src/utils/transactionUtils/transactionUtils.ts (hunk 3 above)
Comment (dtfiedler) [reply src/utils/transactionUtils/transactionUtils.ts]: [27369b5](https://github.com/ar-io/pdns-react/pull/214/commits/27369b5c592e166ae1ea6dd9e9808e1d426eb027)

File: src/components/modals/ManagePDNTModal/ManagePDNTModal.tsx (hunk 1 above)
Comment (dtfiedler) [reply src/components/modals/ManagePDNTModal/ManagePDNTModal.tsx]: [c1aa36b](https://github.com/ar-io/pdns-react/pull/214/commits/c1aa36b05878ee9ebbe3c71fe368154b5a1ba4d9)

File: src/utils/transactionUtils/transactionUtils.ts [hunk 5]
```diff
@@ -588,3 +623,35 @@ export function validateTTLSeconds(ttl: number) {
...
+}[] {
+  // find all pending interactions for the contract, find relevant ones related to row attributes
+  const pendingTxRowData = [];
+  for (const i of pendingContractInteractions) {
+    const attributes = getAttributesFromInteractionFunction(i.payload.function);
+    // TODO: this is not pretty, and could be avoided if we rework the ANT contract to allow `setTTL` and `setTransaction` rather than all of them
+    // relying only on setRecord.
+    for (const attribute of attributes) {
+      // TODO: change targetID to be transactionId
+      const attr = attribute === 'targetID' ? 'transactionId' : attribute;
```
Comment (atticusofsparta) [inline src/utils/transactionUtils/transactionUtils.ts:644]: need something similar to the controller field 
image
owner as well
image

Comment (dtfiedler) [reply src/utils/transactionUtils/transactionUtils.ts:644]: added a util function for these and put a TODO

**Review by atticusofsparta — APPROVED** [review]

**Pull request merged**